plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java

```diff
@@ -183,7 +206,7 @@ public void performAction(IncrementalTaskInputs inputs) throws Exception {
 			if (getCacheFile().exists()) {
 				LastApply lastApply = SerializableMisc.fromFile(LastApply.class, getCacheFile());
 				for (File file : lastApply.changedFiles) {
-					if (!outOfDate.contains(file) && file.exists() && Iterables.contains(target, file)) {
+					if (!outOfDate.contains(file) && file.exists() && Iterables.contains(target, file) && shouldInclude.test(file)) {
```

Comment:
Again, your predicate is probably the fastest one, so I would put it first.  `Iterables.contains(target, file)` can be pretty slow, you should definitely go before that.

Reply:
updated